internal/pkg/stream/ecs.go

```diff
@@ -123,6 +132,18 @@ func (s *ECSDeploymentStreamer) Fetch() (next time.Time, done bool, err error) {
 		return next, false, fmt.Errorf("fetch service description: %w", err)
 	}
 	s.retries = 0
```

Comment:
hmm I wonder if the retries for `ecs` vs `cw` should be separate? (e.g. `ecsRetries` and `cwRetries`)

> DescribeServices:  Bucket refill rate (or Sustained rate) 20
> https://docs.aws.amazon.com/AmazonECS/latest/APIReference/request-throttling.html

> DescribeAlarms: 9 transactions per second (TPS) per Region. The maximum number of operation requests you can make per second without being throttled.
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_limits.html

With the existing implementation, say the following sequence of events happens:
1. We start with `retries: 0`, ECS call succeeds but CW gives a throttling error.
2. We are now at `retries: 1`, ECS call succeeds so we **reset** retries and call CW which gives throttling error again.
3. Now `retries` is back to 1 instead of having 2 for CW.

Reply:
The gist is right now, `nextFetchDate` might be returning an earlier date and we should be penalizing the client more